### PR TITLE
Initial PR for NooBaa plugin package

### PR DESCRIPTION
**Pull request by cloudbehl**
Initial structure PR for NooBaa Plugin package.

### CODE REVIEW

File: frontend/packages/noobaa-storage-plugin/src/models.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  crd: true,
+};
+
+export const NooBaaBackingStoreModel: K8sKind = {
+  label: 'Backing Store',
+  labelPlural: 'Backing Stores',
+  apiVersion: 'v1beta1',
+  apiGroup: 'noobaa.rook.io',
+  plural: 'noobaabackingstores',
+  abbr: 'nbstore',
```
Comment (spadgett) [inline frontend/packages/noobaa-storage-plugin/src/models.ts]: `abbr` should only be 3 or maybe 4 characters and all caps

File: frontend/packages/noobaa-storage-plugin/package.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@console/noobaa-storage-plugin",
+  "version": "0.0.0-fixed",
+  "description": "Noobaa Storage - TDB",
```
Comment (spadgett) [inline frontend/packages/noobaa-storage-plugin/package.json]: Let's not leave TBD in there... I'd just use "Noobaa Storage" for now

Comment (nimrod-becker) [reply frontend/packages/noobaa-storage-plugin/package.json]: NooBaa Storage - Object data service for Kubernetes hybrid and multi-cloud environments

File: frontend/packages/noobaa-storage-plugin/src/models.ts (hunk 1 above)
Comment (nimrod-becker) [reply frontend/packages/noobaa-storage-plugin/src/models.ts]: Can we remove the abbr and not include the field? What is the use of abbr in this flow? We are using it as a shortcut for kubectl...

File: frontend/packages/noobaa-storage-plugin/package.json (hunk 2 above)
Comment (cloudbehl) [reply frontend/packages/noobaa-storage-plugin/package.json]: Update the description as per the suggestion.

File: frontend/packages/noobaa-storage-plugin/src/models.ts (hunk 1 above)
Comment (spadgett) [reply frontend/packages/noobaa-storage-plugin/src/models.ts]: The property might have a confusing name, but it's used in console for the resource type badges by the resource names. It's not related to the kubectl abbreviation. More than 3 or 4 characters doesn't work well.

Search · OKD 2019-06-20 09-07-54

Comment (nimrod-becker) [reply frontend/packages/noobaa-storage-plugin/src/models.ts]: Right, makes sense now.
So we can change to NB for the 'NooBaa System', NBST for the 'Backing Store' and NBC for the 'Bucket Classe'
What do you think?

Comment (vojtechszocs) [reply frontend/packages/noobaa-storage-plugin/src/models.ts]: I'd suggest `NB` for system, `NBS` for backing store and `NBC` for bucket class.

**Comment by cloudbehl** [comment]
@vojtechszocs @spadgett Made the changes requested by @vojtechszocs  in https://github.com/openshift/console/pull/1764#discussion_r295916876 comment.

**Pull request merged**